model.py

```diff
@@ -830,17 +900,23 @@ def FindBestSplit(self,
         probabilities.append(prob)
     return ExponentialMechanism(probabilities, max_gain)
 
-  def GetLeafPrediction(self, gradients: np.array) -> float:
+  def GetLeafPrediction(self, gradients: np.array, y: np.ndarray) -> float:
     """Compute the leaf prediction.
 
     Args:
       gradients (np.array): The gradients for the dataset instances.
+      y (np.ndarray): The dataset labels.
 
     Returns:
       float: The prediction for the leaf node
     """
-    prediction = (-1 * np.sum(gradients) / (len(
-        gradients) + self.l2_lambda))  # type: float
+    if self.loss.is_multi_class and len(gradients) > 0:
+      prediction = -1 * np.sum(gradients) * (self.loss.K - 1) / self.loss.K
```

Comment:
Can you link here a paper / documentation that explains the reasoning for the predictions formula? Just for reference.